### PR TITLE
handle cases where containers restart

### PR DESCRIPTION
**Pull request by deads2k**
When containers restart and do not end, the container lifecycle continues.  This fixes that logic.

### CODE REVIEW

File: pkg/monitor/event.go [hunk 1]
```diff
@@ -123,18 +123,7 @@ func recordAddOrUpdateEvent(
 	// special case some very common events
 	switch obj.Reason {
 	case "":
-	case "Scheduled":
```
Comment (deads2k) [inline pkg/monitor/event.go:126]: This was duplicated

File: pkg/monitor/event.go [hunk 2]
```diff
@@ -123,18 +123,7 @@ func recordAddOrUpdateEvent(
...
-			if strings.HasPrefix(message, "Successfully assigned ") {
-				if i := strings.Index(message, " to "); i != -1 {
-					node := message[i+4:]
-					message = fmt.Sprintf("node/%s reason/%s", node, obj.Reason)
-					break
-				}
-			}
-		}
-		message = fmt.Sprintf("reason/%s %s", obj.Reason, message)
-	case "Started", "Created", "Killing":
```
Comment (deads2k) [inline pkg/monitor/event.go:137]: so was started and created here.

File: pkg/monitor/intervalcreation/pod.go [hunk 3]
```diff
@@ -305,11 +305,11 @@ func (t containerLifecycleTimeBounder) getEndTime(inLocator string) time.Time {
 	if !ok {
 		return t.delegate.getEndTime(locator)
 	}
-	for i := len(containerEvents) - 1; i >= 0; i-- {
-		event := containerEvents[i]
-		if monitorapi.ReasonFrom(event.Message) == monitorapi.ContainerReasonContainerExit {
-			return event.From
-		}
+	// if the last event is a containerExit, then that's as long as the container lasted.
```
Comment (deads2k) [inline pkg/monitor/intervalcreation/pod.go:308]: this is the core fix.

File: pkg/monitor/pod.go [hunk 4]
```diff
@@ -597,45 +569,14 @@ func lastContainerTimeFromStatus(current *corev1.ContainerStatus) time.Time {
 	return time.Time{}
 }
 
-func conditionsForTransitioningContainer(pod *corev1.Pod, current, previous *corev1.ContainerStatus, init bool, reason, cause, message string, currentTime time.Time, lastContainerTime time.Time) []monitorapi.Condition {
```
Comment (deads2k) [inline pkg/monitor/pod.go:600]: the different cases here didn't provide value. They exist from before we started building a history of pod lifecycle in true intervals after the instants were reported.

File: pkg/monitor/intervalcreation/pod.go [hunk 5]
```diff
@@ -305,11 +305,11 @@ func (t containerLifecycleTimeBounder) getEndTime(inLocator string) time.Time {
...
+	// if the last event isn't a containerExit, then the last time we're aware of for the container is parent.
```
Comment (DennisPeriquet) [inline pkg/monitor/intervalcreation/pod.go:309]: you mean `for the container's parent` right?

**Comment by deads2k** [comment]
built. this improves reporting, merging.

**Pull request merged**